plugins/woocommerce-admin/client/launch-your-store/tour/use-site-visibility-tour.tsx

```diff
@@ -31,15 +31,18 @@ export const useSiteVisibilityTour = () => {
 		};
 	} );
 
-	const onClose = () => {
+	const onClose = async () => {
 		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
 		// @ts-ignore
-		dispatch( 'core' ).saveUser( {
+		await dispatch( 'core' ).saveUser( {
 			id: window?.wcSettings?.currentUserId,
 			meta: {
 				woocommerce_launch_your_store_tour_hidden: 'yes',
 			},
 		} );
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-ignore
+		dispatch( 'core' ).invalidateResolution( 'getCurrentUser' );
```

Comment:
Interesting that you'd need to invalidate the last resolution here. Any idea why `shouldTourBeShown` from the `useSelect` isn't reading the latest data in its `getCurrentUser` call?

Reply:
I think it's related to how the data flow and cache management work in wp core data. 🤔  The `saveUser` call doesn't update the local currentUser state after the user is saved to the server. So, the `shouldTourBeShown` function is reading the old data from the cache.

https://github.com/user-attachments/assets/b49b0bfc-342f-4ae9-b8a5-dd24010f69d1

![Screenshot 2024-08-13 at 16 25 31](https://github.com/user-attachments/assets/e6b0a885-8d95-4d03-91cc-da80d389e8ac)